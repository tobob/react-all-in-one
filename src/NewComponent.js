import React, { Component } from "react";

export default class NewComponent extends Component {
  constructor(props) {
    super(props);

    if (props.callback) {
      props.callback();
    }

    this.state = {
      clicked: 0
    };
  }

  handleEvent(event) {
    this.setState(prevState => ({ clicked: prevState.clicked + 1 }));
  }

  render() {
    return (
      <>
        <h1>New Component</h1>
        <button onClick={e => this.handleEvent(e)}>{this.state.clicked}</button>
      </>
    );
  }
}
