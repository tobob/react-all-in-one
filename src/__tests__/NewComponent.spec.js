import React from "react";
import NewComponent from "../NewComponent";
import { shallow } from "enzyme";

let props;

describe("NewComponet", () => {
  beforeEach(() => {
    props = {
      something: "koko",
      callback: jest.fn()
    };
  });

  it("renders well", () => {
    props.something = 1;
    const wrapper = shallow(<NewComponent {...props} />);
    console.log(wrapper);

    expect(wrapper.find("button").text()).toEqual("0");
    expect(props.callback).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });

  it("keeps counter state initialy", () => {
    const wrapper = shallow(<NewComponent />);

    expect(wrapper.state("clicked")).toEqual(0);
  });

  it("increase clicked value on onClick event handler", () => {
    const wrapper = shallow(<NewComponent />);

    const aaa = jest.fn();

    wrapper.instance().handleEvent = aaa;
    const button = wrapper.find("button");
    button.simulate("click");
    expect(aaa).toHaveBeenCalled();
  });

  it("increase clicked value on onClick event handler", () => {
    const wrapper = shallow(<NewComponent />);
    const button = wrapper.find("button");
    button.simulate("click");
    expect(wrapper.state("clicked")).toEqual(1);
  });
});
