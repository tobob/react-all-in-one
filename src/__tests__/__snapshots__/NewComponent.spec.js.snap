// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewComponet renders well 1`] = `
<Fragment>
  <h1>
    New Component
  </h1>
  <button
    onClick={[Function]}
  >
    0
  </button>
</Fragment>
`;
